experiments/AtmosLES/stable_bl_les.jl

```diff
@@ -0,0 +1,117 @@
+include("stable_bl_model.jl")
+
+function main()
+
+    # TODO: this will move to the future namelist functionality
+    sbl_args = ArgParseSettings(autofix_names = true)
+    add_arg_group!(sbl_args, "StableBoundaryLayer")
+    @add_arg_table! sbl_args begin
+        "--surface-flux"
+        help = "specify surface flux for energy and moisture"
+        metavar = "prescribed|bulk"
+        arg_type = String
+        default = "bulk"
+    end
+
+    cl_args = ClimateMachine.init(parse_clargs = true, custom_clargs = sbl_args)
+
+    surface_flux = cl_args["surface_flux"]
+
+    ClimateMachine.init(parse_clargs = true)
+
+    FT = Float64
+    config_type = AtmosLESConfigType
+
+    # DG polynomial order
+    N = 4
+    # Domain resolution and size
+    Δh = FT(20)
+    Δv = FT(20)
+
+    resolution = (Δh, Δh, Δv)
+
+    # Prescribe domain parameters
+    xmax = FT(100)
+    ymax = FT(100)
+    zmax = FT(400)
+
+    t0 = FT(0)
+
+    # Required simulation time == 9hours
+    timeend = FT(3600 * 0.1)
+    CFLmax = FT(0.4)
+
+    # Choose default IMEX solver
+    ode_solver_type = ClimateMachine.IMEXSolverType()
+
+    model = stable_bl_model(FT, config_type, zmax, surface_flux)
+    ics = model.problem.init_state_prognostic
+
+    # Assemble configuration
+    driver_config = ClimateMachine.AtmosLESConfiguration(
+        "StableBoundaryLayer",
+        N,
+        resolution,
+        xmax,
+        ymax,
+        zmax,
+        param_set,
+        init_problem!,
+        solver_type = ode_solver_type,
+        model = model,
+    )
+
+    solver_config = ClimateMachine.SolverConfiguration(
+        t0,
+        timeend,
+        driver_config,
+        init_on_cpu = true,
+        Courant_number = CFLmax,
+    )
+    dgn_config = config_diagnostics(driver_config)
+
+    cbtmarfilter = GenericCallbacks.EveryXSimulationSteps(1) do
+        Filters.apply!(
+            solver_config.Q,
+            ("moisture.ρq_tot",),
+            solver_config.dg.grid,
+            TMARFilter(),
+        )
+        nothing
+    end
+
+    # State variable
+    Q = solver_config.Q
+    # Volume geometry information
+    vgeo = driver_config.grid.vgeo
+    M = vgeo[:, Grids._M, :]
+    # Unpack prognostic vars
+    ρ₀ = Q.ρ
+    ρe₀ = Q.ρe
+    # DG variable sums
+    Σρ₀ = sum(ρ₀ .* M)
+    Σρe₀ = sum(ρe₀ .* M)
+    cb_check_cons = GenericCallbacks.EveryXSimulationSteps(3000) do
+        Q = solver_config.Q
+        δρ = (sum(Q.ρ .* M) - Σρ₀) / Σρ₀
+        δρe = (sum(Q.ρe .* M) .- Σρe₀) ./ Σρe₀
+        @show (abs(δρ))
+        @show (abs(δρe))
+        nothing
+    end
```

Comment:
```suggestion
```
I believe that the `check_cons = ...` below does exactly this (I think it's the new interface, right @kpamnany?)

Reply:
So should I delete this part, and leave the `user_callback`?

Reply:
That is my understanding.